config/v1/types_cluster_monitoring.go

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ClusterMonitoring is the Custom Resource object which holds the current status of Cluster Monitoring Operator. CMO is a central component of the monitoring stack.
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:internal
+// +openshift:api-approved.openshift.io=https://github.com/openshift/api/pull/1929
+// +openshift:file-pattern=cvoRunLevel=0000_10,operatorName=config-operator,operatorOrdering=01
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:path=clustermonitoring,scope=Cluster
```

Comment:
It is used in HyperShift

Reply:
Do you know on which side it is used? Is it on the management or guest cluster side?

Reply:
Not sure about it.  @simonpasquier do you have this info?

Reply:
Iiuc, this should be used on the management cluster side, as this will eventually contain the settings for cluster admins. The guest clusters would rather use the `UserDefinedMonitoring` struct.

Reply:
I think we need to have a discussion with the HyperShift folks about this API then.

If this is cluster scoped, then it makes sense in the OCP world.

If it is namespace scoped, the it makes sense in HyperShift.

The converse of both of my previous statements are not true.

To be correct with our APIs, it would be preferable to have a `Hosted<CRD>` that is namespaced, and exists only in the management cluster, and then have no semblance of the feature exist within the guest cluster. 

This pattern is coming up a lot at the moment with new designs for CRDs, will need to come up with a common path forward

CC @enxebre

Reply:
any news about it?

Reply:
Having similar discussions in https://github.com/openshift/enhancements/pull/1571 about using a Hosted version of a CRD to prevent the issues we have where we have discrepancies between standalone and hosted

There's also one about CVO log levels that has taken a different approach

Reply:
Did you have any discussion with the HyperShift folks about how to handle this? I believe someone was setting up a topic for their office hours?

Reply:
Yep https://issues.redhat.com/browse/HOSTEDCP-2194

Reply:
hcp exposes selected v1config knobs in the [HostedCluster API](https://github.com/openshift/hypershift/blob/main/api/hypershift/v1beta1/hostedcluster_types.go#L537) It would seem reasonable to expose intent for this there as well if we need to.  API Consumers would need to assess coverage and intersection of existing scenarios, https://github.com/openshift/hypershift-dataplane-metrics-forwarder/blob/main/hack/templates/00-hypershift-dataplane-metrics-forwarder.SelectorSyncSet.yaml.tmpl#L107

Now, re how this input propagates down (namespace vs cluster scope): while making it namespaced will give more flexibility for future pivots I don't think there's a current direction or effort to move the cluster monitoring into the management side. cc @csrwng @typeid

Reply:
For ROSA HCP, our ultimate goal is to fully monitor the platform purely from the management cluster side via metrics exposed by the hypershift/control-plane components.

We have [XCMSTRAT-976](https://issues.redhat.com/browse/XCMSTRAT-976) to hand off CMO completely to the user, [up to the point that they can end up enabling/disabling it as a whole](https://issues.redhat.com/browse/MON-3152), therefore I think it makes sense to keep it as is (CMO configurable from the hosted cluster side) - feel free to reach out in that XCMSTRAT in case more clarification is needed.

Reply:
Ok, so if I'm reading the last few comments correctly, we are happy to proceed with this design as is, and will create a new abstraction at the HostedCluster level for HCP relevant parts of monitoring the management side